README.md

```diff
@@ -47,299 +47,477 @@ The FWS Payments contract enables ERC20 token payment flows through "rails" - au
+-   **Modify Rail Terms (`modifyRailLockup`, `modifyRailPayment`):** Adjust the payment rate, lockup period, and fixed lockup amount for any rail it manages. Any increase in the payer's financial commitment is checked against the operator's allowances.
+-   **Execute One-Time Payments (`modifyRailPayment`):** Execute one-time payments from the rail's fixed lockup.
+-   **Settle Rails (`settleRail`):** Trigger payment settlement for a rail to process due payments within the existing terms of the rail. As a rail participant, the operator can initiate settlement at any time. The operator cannot, however, arbitrarily settle a rail for a higher-than-expected amount or higher than expected duration.
+-   **Terminate Rails (`terminateRail`):** End a payment rail. Unlike payers, an operator can terminate a rail even if the payer's account is not fully funded.
 
-Each payment rail requires the user to manage two lockup components:
+### Per-Rail Lockup: The Guarantee Mechanism
 
-- **Payment Stream lockup:** `paymentRate × lockupPeriod` (covers future payment streams)
-- **Fixed lockup:** `lockupFixed` (covers one-time payments)
+Each payment rail can be configured to require the payer to lock funds to guarantee future payments. This lockup is composed of two distinct components:
 
-The contract always enforces that the sum of these two values is locked for each rail:
+-   **Streaming Lockup (`paymentRate × lockupPeriod`):** A calculated guarantee for rate based payments for a pre-agreed lockup period.
+-   **Fixed Lockup (`lockupFixed`):** A specific amount set aside for one-time payments.
 
-```
-Total Lockup Required = (paymentRate × lockupPeriod) + lockupFixed
-```
+The total lockup for a payer's account is the sum of these requirements across *all* their active rails. This total is reserved from their deposited funds and cannot be withdrawn.
+
+#### The Crucial Role of Streaming Lockup: A Safety Hatch, Not a Pre-payment
+
+It is critical to understand that the streaming lockup is **not** a pre-paid account that is drawn from during normal operation. Instead, it functions as a **safety hatch** that can only be fully utilized *after* a rail is terminated.
+
+**1. During Normal Operation (Before Termination)**
 
-- You must ensure both components are sufficiently funded.
-- One-time payments are deducted from `lockupFixed`.
-- Streaming payments are covered by the streaming lockup.
+While a rail is active, the streaming lockup acts as a **guarantee of solvency for a pre-agreed number of epochs**, not as a direct source of payment.
 
-#### How Each Lockup Bucket Works
+-   **Payments from General Funds:** When `settleRail` is called on an active rail, payments are drawn from the payer's general `funds`.
+-   **Lockup as a Floor:** The lockup simply acts as a minimum balance. The contract prevents the payer from withdrawing funds below this floor.
```

Comment:
In case we want to have a pre-payment type of service then we'd need to:

- Set lockup fixed to some N amount of tokens
- Have user deposit N amount of tokens to their account, updating the account `funds` to equal the deposited amount 

Is this correct?

Reply:
@pyropy The call to set the fixed lockup i.e. `modifyTerminatedRailLockup` will fail if the user does not already enough funds to be locked.